kernel/common/enumerator.rb

```diff
@@ -148,7 +149,19 @@ def rewind
     end
 
     def size
-      @size.kind_of?(Proc) ? @size.call : @size
+      if @size.kind_of?(Proc)
+        @size.call
+      elsif @size.is_a? Numeric
+        @size
+      elsif @size.respond_to? :size
+        @size.size
+      else
```

Comment:
You mean `size` was returning an Array? If so, that should probably belong in its own PR (unless it's directly related to `each_slice`).